File: src/MileageTracker.WebAPI/Models/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Identity;

namespace MileageTracker.WebAPI.Models
{
    [Table("Address")]
    public class Address
    {
        [Key]
        public int AddressId { get; set; }
        [Display(Name = "Address Line One")]
        public string AddressLineOne { get; set; }
        [Display(Name = "Address Line Two")]
        public string AddressLineTwo { get; set; }
        [Display(Name = "City")]
        public string City { get; set; }
        [Display(Name = "State Abbreviation")]
        public string StateAbbreviation { get; set; }
        [Display(Name = "Zip Code")]
        public int ZipCode { get; set; }
        [ForeignKey(nameof(User))]
        public string UserId { get; set; }
        public IdentityUser User { get; set; }
    }
}

File: src/client/JS/Address.js
var uri = "https://localhost:5001/api/Address";

function CreateAddress() {
    $(`[name="addAddressLineOne"]`).empty();
    $(`[name="addCity"]`).empty();
    $(`[name="addState"]`).empty();
    $(`[name="addZipCode"]`).empty();

    const address = {
        AddressLineOne: $(`#add-address-line-one`).val(),
        AddressLineTwo: $(`#add-address-line-two`).val(),
        City: $(`#add-city`).val(),
        StateAbbreviation: $(`#add-state`).val(),
        ZipCode: $(`#add-zip-code`).val()
    };

    if (address.AddressLineOne === "") {
        $(`[name="addAddressLineOne"]`).append("* Address Line One is required.");
        return;
    } else if (address.City === "") {
        $(`[name="addCity"]`).append("* City is required.");
    } else if (address.StateAbbreviation === "") {
        $(`[name="addState"]`).append("* State is required.");
    } else if (address.ZipCode === "") {
        $(`[name="addZipCode"]`).append("* Zip code is required.");
    } else {
        $.ajax({
            type: "POST",
            accepts: "application/json",
            url: uri,
            contentType: "application/json",
            data: JSON.stringify(address),
            error: error => {
                console.log(error);
            },
            success: result => {
                $(`#add-address-line-one`).val("");
                $(`#add-address-line-two`).val("");
                $(`#add-city`).val("");
                $(`#add-state`).val("");
                $(`#add-zip-code`).val("");               
            }
        });
    }
}